frontend/src/APIClients/AuthAPIClient.ts

```diff
@@ -6,21 +11,35 @@ import {
   setLocalStorageObjProperty,
 } from "../utils/LocalStorageUtils";
 
+type LoginFunction = (
```

Comment:
oof. Definitely, a moment where typescript feels like a mega pain in the butt. 
Played around on your branch and found that this is more concise and worked:
```
type LoginFunction = (
  options: MutationFunctionOptions<
    { login: AuthenticatedUser },
    OperationVariables
  >,
) => Promise<FetchResult<{ login: AuthenticatedUser }>>;
```

Reply:
yeah, that looks better! I got this straight from starter-code so might be worth noting the change for the original repo?

Reply:
ah good to know! I'll mention it to the internal tools team